system/jlib/jfile.cpp

```diff
@@ -6219,6 +6219,57 @@ IReplicatedFile *createReplicatedFile()
 }
 
 
+// ---------------------------------------------------------------------------------
+
+class CSerialInputStreamBase : public CInterfaceOf<ISerialInputStream>
+{
+
+};
+class CTallyInputStream : public CSerialInputStreamBase
+{
+public:
+    CTallyInputStream(Linked<ISerialInputStream> _in, IFileSerialStreamCallback & _tally)
+    : in(_in), tally(_tally)
+    {
+    }
+
+    virtual size32_t read(size32_t len, void * ptr) override
+    {
+        offset_t pos = in->tell();
+        size32_t ret = in->read(len, ptr);
+        tally.process(pos, ret, ptr);
+        return ret;
+
+    }
+    virtual void skip(size32_t sz) override
+    {
+        throwUnexpectedX("Skip called on an input stream that is being tallied");
+        in->skip(sz);
+    }
+    virtual void get(size32_t len, void * ptr)
+    {
+        offset_t pos = in->tell();
+        in->get(len, ptr);
+        tally.process(pos, len, ptr);
+    }
+    virtual bool eos()
+    {
+        return in->eos();
+    }
+    virtual void reset(offset_t _offset, offset_t _flen)
+    {
+        in->reset(_offset, _flen);
```

Comment:
Question: Should the CRC be (somehow) modified/adjusted/discarded when reset() is called? IFileSerialStreamCallback does not support anything like that, but should it?